.github/workflows/check-dev-containers.yml

```diff
@@ -0,0 +1,56 @@
+name: Check Dev Containers
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+  workflow_call:
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.head_ref || github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  changes:
```

Comment:
If we are splitting the build steps for performance, why not splitting the checks too?

Reply:
I unzip that comment for me:

The time for checking is building the action and the checkout. The check itself is fast.

Thus, when moving the changes-check to the respective job, we gain performance in three ways:

1. avoidance of one checkout call
2. avoidance of starting another job by GitHub workflows
3. better maintenance, because cross-job dependencies are harder to maintain then inside-job dependencies (not scientifically proofed statement ^^)